core/src/main/java/io/seata/core/rpc/netty/NettyClientChannelManager.java

```diff
@@ -221,22 +227,84 @@ private Channel doConnect(String serverAddress) {
             channelFromPool = nettyClientKeyPool.borrowObject(poolKeyMap.get(serverAddress));
             channels.put(serverAddress, channelFromPool);
         } catch (Exception exx) {
-            LOGGER.error("{} register RM failed.",FrameworkErrorCode.RegisterRM.getErrCode(), exx);
+            LOGGER.error("{} register RM failed.", FrameworkErrorCode.RegisterRM.getErrCode(), exx);
             throw new FrameworkException("can not register RM,err:" + exx.getMessage());
         }
         return channelFromPool;
     }
 
     private List<String> getAvailServerList(String transactionServiceGroup) throws Exception {
         List<InetSocketAddress> availInetSocketAddressList = RegistryFactory.getInstance()
-                                                                            .lookup(transactionServiceGroup);
+                .lookup(transactionServiceGroup);
+
+        List<InetSocketAddress> currentInetSocketAddressList = RegistryFactory.getInstance()
+                .aliveLookup(transactionServiceGroup);
         if (CollectionUtils.isEmpty(availInetSocketAddressList)) {
+            if (CollectionUtils.isEmpty(currentInetSocketAddressList)) {
+                currentInetSocketAddressList.clear();
+            }
             return Collections.emptyList();
         }
+        for (InetSocketAddress address : availInetSocketAddressList) {
+            boolean canConnect = false;
+            for (int tryCount = 0; tryCount < TRY_CONNECT_COUNT; tryCount++) {
+                if (isServerAddressConnect(address.toString())) {
+                    canConnect = true;
+                    break;
+                }
+            }
+            if (canConnect) {
+                if (!currentInetSocketAddressList.contains(address)) {
+                    currentInetSocketAddressList.add(address);
+                }
+            } else {
+                if (currentInetSocketAddressList.contains(address)) {
```

Comment:
I suggest remove this, because remove already include constant operation and cause two O(N) operation.